cmd/frontend/internal/app/ui/handlers.go

```diff
@@ -256,6 +257,75 @@ func serveSignIn(w http.ResponseWriter, r *http.Request) error {
 	return renderTemplate(w, "app.html", common)
 }
 
+// redirectTreeOrBlob redirects a blob page to a tree page if the file is actually a directory,
+// or a tree page to a blob page if the directory is actually a file.
+func redirectTreeOrBlob(routeName string, common *Common, r *http.Request, w http.ResponseWriter) (requestHandled bool, err error) {
+	path := mux.Vars(r)["Path"]
+	if path == "/" || path == "" {
+		if routeName != routeRepo {
+			// Redirect to repo route
+			target := "/" + string(common.Repo.Name)
+			if common.Rev != "" {
+				target += common.Rev
+			}
```

Comment:
nit: If the `common.Rev` is empty, I guess it doesn't hurt.
```suggestion
			target := "/" + string(common.Repo.Name) + common.Rev
```